flang/lib/Lower/OpenMP/OpenMP.cpp

```diff
@@ -2263,24 +2321,13 @@ static void genOMPDispatch(lower::AbstractConverter &converter,
 
   // Composite constructs
   case llvm::omp::Directive::OMPD_distribute_parallel_do:
-    genCompositeDistributeParallelDo(converter, symTable, semaCtx, eval, loc,
-                                     queue, item, *loopDsp);
-    break;
   case llvm::omp::Directive::OMPD_distribute_parallel_do_simd:
-    genCompositeDistributeParallelDoSimd(converter, symTable, semaCtx, eval,
-                                         loc, queue, item, *loopDsp);
-    break;
   case llvm::omp::Directive::OMPD_distribute_simd:
-    genCompositeDistributeSimd(converter, symTable, semaCtx, eval, loc, queue,
-                               item, *loopDsp);
-    break;
   case llvm::omp::Directive::OMPD_do_simd:
-    genCompositeDoSimd(converter, symTable, semaCtx, eval, loc, queue, item,
-                       *loopDsp);
-    break;
   case llvm::omp::Directive::OMPD_taskloop_simd:
-    genCompositeTaskloopSimd(converter, symTable, semaCtx, eval, loc, queue,
-                             item, *loopDsp);
+    // Composite constructs should have been split into a sequence of leaf
+    // constructs and lowered by genOMPCompositeDispatch().
+    llvm_unreachable("Unexpected composite construct.");
     break;
   default:
```

Comment:
Maybe you could delete the specific cases with composite constructs, and add `assert(!isCompositeConstruct(dir))` to the default label.

Reply:
Done. I added a bit more generic assert, since we don't expect combined constructs there either.